scripts/tools/metricProgrammable.yml

```diff
@@ -0,0 +1,490 @@
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Create metric group handle."
+version: "1.8"
+class: $tMetricGroup
+name: CreateExp
+decl: static
+details:
+    - "Metric handles from multiple domains may be used for a single metric group."
+    - "Metric handles from different sourceIds (refer $t_metric_programmable_exp_properties_t) are not allowed in a single metric group."
+params:
+    - type: $t_device_handle_t
+      name: hDevice
+      desc: "[in] handle of the device"
+    - type: const char
+      name: name[ZET_MAX_METRIC_GROUP_NAME]
+      desc: "[in] Metric group name."
+    - type: const char
+      name: description[ZET_MAX_METRIC_GROUP_DESCRIPTION]
+      desc: "[in] Metric group description."
+    - type: $t_metric_group_handle_t*
+      name: phMetricGroup
+      desc: "[out] Created Metric group handle"
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Add a metric handle to the metric group handle created using $tMetricGroupCreateExp."
+version: "1.8"
+class: $tMetricGroup
+name: AddMetricExp
+decl: static
+details:
+    - "Reasons for failing to add the metric could be queried using pErrorString"
+params:
+    - type: $t_metric_group_handle_t 
+      name: hMetricGroup
+      desc: "[in] Handle of the metric group"
+    - type: $t_metric_handle_t
+      name: hMetric
+      desc: "[in] Metric to be added to the group."
+    - type: "uint32_t*"
+      name: pErrorStringSize
+      desc: "[in,out][optional] Size of the error string to query, if an error was reported during adding the metric handle."
+    - type: "char*"
+      name: pErrorString
+      desc: |
+            [in,out][optional][range(0, *pErrorStringSize)] Error string.
+            if *pErrorStringSize is less than the length of the error string available, then driver shall only retrieve that length of error string.
+returns:
+    - $X_RESULT_ERROR_INVALID_ARGUMENT:
+        - "If a Metric handle from a pre-defined metric group is requested to be added"
+    - $X_RESULT_ERROR_UNSUPPORTED_FEATURE:
+        - "If the Metric group handle is already activated"
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Remove a metric from the metric group handle created using $tMetricGroupCreateExp."
+version: "1.8"
+class: $tMetricGroup
+name: RemoveMetricExp
+decl: static
+details:
+    - "Remove an already added metric handle from the metric group"
+params:
+    - type: $t_metric_group_handle_t
+      name: hMetricGroup
+      desc: "[in] Handle of the metric group"
+    - type: $t_metric_handle_t
+      name: hMetric
+      desc: "[in] Metric handle to be removed from the metric group."
+returns:
+    - $X_RESULT_ERROR_INVALID_ARGUMENT:
+        - "If trying to remove a metric not previously added to the metric group"
```

Comment:
how about calling it on an active metric group?

Reply:
should this description be added?

Reply:
Added it